### PR TITLE
Minor cleanup to FlutterLogo.

### PR DESCRIPTION
**Pull request by Hixie**
This started as the solution to https://github.com/flutter/flutter/issues/12005 but there was nothing to do so all I had left was cleanup.

### CODE REVIEW

File: packages/flutter/lib/src/painting/flutter_logo.dart [hunk 1]
```diff
@@ -47,11 +47,16 @@ class FlutterLogoDecoration extends Decoration {
     this.textColor: const Color(0xFF616161),
     this.style: FlutterLogoStyle.markOnly,
     this.margin: EdgeInsets.zero,
-  }) : _position = style == FlutterLogoStyle.markOnly ? 0.0 : style == FlutterLogoStyle.horizontal ? 1.0 : -1.0, // ignore: CONST_EVAL_TYPE_BOOL_NUM_STRING
+  }) : assert(lightColor != null),
+       assert(darkColor != null),
+       assert(textColor != null),
+       assert(style != null),
+       assert(margin != null),
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/flutter_logo.dart]: Should the comment above mention that margin also cannot be null?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/flutter_logo.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/painting/flutter_logo.dart]: Yes! Good catch. Fixed. Thanks.

**Comment by Hixie** [comment]
will land on green

**Pull request merged**